docs/codebase-best-practices.md

```diff
@@ -1,5 +1,23 @@
 # Codebase Best Practices
 
+## Styling a component
+
+Be mindful of adding new colors to the codebase, as you will have to account for contrast and color blindness. Because of this, we always use already determined variables in [variable.css](/client/src/components/layouts/variables.css).
+
+We are against adding new variables to [variable.css](/client/src/components/layouts/variables.css), because we have to account for contrast, color blindness 🙂, and cluttering the devtools which affects negatively the devtools experiences and add concerns for accessibility.
+
+Use `!important` for accessibility concerns only after leaving a comment describing the issue, so it won't be removed by mistake in future refactoring.
+
+### RTL support
+
+We are striving to support RTL layout in the codebase, for this you need be mindful of how to style the component in your PR. Here are a quick role of thumbs to follow:
+
+- Don't use float properties: Although it may seem best to have the component in the "perfect position", you will be climbing mountains to reach that perfect positioning in responsive layout, and you will need to reach higher heights to have it supported in RTL. 
+- - Use flexbox and Grid layouts instead, as they have RTL support already built in them, and it will be easier to maintain and review
+- Don't define the direction while using `margin` and `padding`: it may seem harmless to use `padding-right` and `margin-left`, but these directions aren't mirrored when the layout change to RTL, and adding counter values for them in the RTL file makes maintaining the codebase harder.
```

Comment:
```suggestion
- Don't define the direction while using `margin` and `padding`: it may seem harmless to use `padding-right` and `margin-left`, but these directions aren't mirrored when the layout changes to RTL, and adding counter values for them in the RTL file makes maintaining the codebase harder.
```